web_external/views/body/DataPanel.js

```diff
@@ -281,24 +372,64 @@ export default Panel.extend({
         return this.parentView.parentView.model;
     },
 
+    selectDataset(e) {
+        var datasetId = $(e.target).attr('m-dataset-id');
+        if (e.target.checked) {
+            this.selectedDatasetId.add(datasetId);
+        } else {
+            this.selectedDatasetId.delete(datasetId);
+        }
+        this.render();
+    },
+
+    selectCategory(e) {
+        var datasetIds = $(e.target).closest('.m-sub-category').find('.dataset').map((i, el) => $(el).attr('m-dataset-id')).toArray();
```

Comment:
can we wrap this long line? I do not have any strong preference but something < 120 works most of the time. Does Girder has any guideline?

Reply:
Girder has requirements for Python line lengths (100 characters), but not for Javascript.  There are endless debates about line lengths.  I personally wish we *wouldn't* break conceptual lines -- let your editor format it visually to whatever you find handy, rather than forcing this into the code; keeping it together makes it more easily found using a tool like grep.  However, others feel strongly otherwise.  There isn't any correct answer to this.

Reply:
I think in this particular case we could break the line at map() since sometimes the map could have a full function as parameter which would normally on a different line.
PS: I personally always use automatic line wrap in my editor, and the editor always starts the wrapped lines at the proper position (with the same amount of intention as the original line). So I never felt the need to explicitly make a line shorter.